### PR TITLE
Server-side filtering

### PR DESCRIPTION
**Pull request by esheehan-gsl**
Move the filtering behavior out of the client and into the server. This change
will help simplify the client application, making it less error prone. All the
client will have to do is build URLs with parameters based on the selections in
the charts, it doesn't have to implement any filtering logic.

### CODE REVIEW

File: services/ui/src/js/components/Chart2DHistogram/Chart2DHistogram.js [hunk 1]
```diff
@@ -205,6 +221,19 @@ export default class Chart2DHistogram extends ChartElement {
...
+  }
+
+  set src(value) {
+    if (!value) {
+      this.removeAttribute("src");
+      return;
+    }
+
+    this.setAttribute("src", value);
+  }
```
Comment (ian-noaa) [inline services/ui/src/js/components/Chart2DHistogram/Chart2DHistogram.js:235]: A naming question that I had - I think `src` here represents the data from the backend. Is that understanding correct? If so, would `data` be a better name? However, I was also seeing a `data` member in the class. I spent some time trying to figure out what `src` is.

**Review by ian-noaa — APPROVED** [review]
Looks good to me. I think refactoring the filters into their own class/module would make a lot of sense.

File: services/ui/src/js/components/Chart2DHistogram/Chart2DHistogram.js (hunk 1 above)
Comment (esheehan-gsl) [reply services/ui/src/js/components/Chart2DHistogram/Chart2DHistogram.js:235]: That’s a good question. Let me check the JSDoc to make sure this is explained clearly.

The `src` attribute is a URL from which data is fetched. I’m mimicking the same API that replaced HTML elements like `` use. For an image, you do `` and the browser fetches the PNG, parses it, and draws it on screen. For our charts we can do `<chart-2dhistogram src="/api/wind/..."></chart-2dhistogram>` and our web component will use `fetch()` to get the data and render it on screen, the same way the browser does for an image.

The `data` property is a way to load data outside of the component and pass it in. It was something we were using when we were loading the data in our Svelte components and doing some filtering before handing it off to the web component to render.

Comment (esheehan-gsl) [reply services/ui/src/js/components/Chart2DHistogram/Chart2DHistogram.js:236]: Well, not that JSDoc is especially easy to read, especially on a class with this many props, but at least now the `src` attribute is documented.

Comment (ian-noaa) [reply services/ui/src/js/components/Chart2DHistogram/Chart2DHistogram.js:236]: Gotcha. I was wondering if it was just my lack of familiarity with HTML/JS and indeed it was! 😁 That makes a lot more sense. Thanks!

**Pull request merged**